progetto_esa.tex

```diff
@@ -95,39 +116,106 @@ \section{The Compiling Process}
+In our first example we are going to compile the buildroot for qemu virtual machine.
 Def\_config files are invoked before that we are going to start the menuconfig tool, so that the configuration is almost ready for our specific real or virtual platform.
 In this first example we have to invoke the def:config for qemu, as said.
 we simply run the command:
-make qemu\_a\rm\_versatile\_defconfig
+make qemu\_arm\_versatile\_defconfig
 and then, we are ready to start the menu\_config to customize our distribution parameters, like host name or root password.
-After that, we can run the make command and at the end we'll have our .img file ready.
+After that, we can run the make command and at the end we'll have our image files ready.
+All the files resides in the output/ folder.
 
 \section{Compiling for real devices}
-Buildroot is an extremely growing tol that is able to support almost every board known in the market.
-In /board folder we can find the instruction to compile and run buildroot for all the devices supported.
-In particular, we have to consider the readme file that indicates the specific defconfig file to invoke.
+Build root is an extremely growing tool that is able to support almost every board present in the market.
+In /board folder we can find the instructions to compile and run build root for all the devices supported.
+In particular, we have to consider the readme.md file that indicates the specific def\_config file to invoke to customize the compilation process.
+For example, in our case we choose to customize and compile build rott for a very performant and affordable board called OrangePi PC Plus.
+First, we are going to invoke the def\_config file as follow:
+make orangepi\_pc\_plus\_defconfig and then we are ready to start the compiling process.
+After that, once the compiling process is completed you will have an image called "sdcard.img"
+in the output/images/ directory.
+At this point, you are ready copy the bootable "sdcard.img" onto an SD card with dd command as follow:
+dd if=output/images/sdcard.img of=/dev/sdX
+where sdX is the device identifier  where you sd card is configured.
 
 \section{Customizing and adding packages to buildroot}
+Buildroot is highly customizable and it allows to integrate in your distribution every kind of packages and kernel modules.
 you can place project-specific customizations in two locations:
-
-directly within the Buildroot tree, typically maintaining them using branches in a version control system so that upgrading to a newer Buildroot release is easy.
-outside of the Buildroot tree, using the br2-external mechanism. 
-This mechanism allows to keep package recipes, board support and configuration files outside of the Buildroot tree, while still having them nicely integrated in the build logic. 
+\begin{itemize}
+\item Directly within the Buildroot tree, typically maintaining them using branches in a version control system so that upgrading to a newer Buildroot release is easy.
```

Comment:
This is doable but discouraged. If you have a buildroot branch you have actually a fork. Forking means that you need to maintain that fork by rebasing often from the master branch. This can be very difficult to do if your branch has diverged from the master in a sensible way.